### DISCUSSION TITLE
Missing setupx_ ILI9486.h  for Wemos D1 R32 with a 3.5"tft shield (controller ILI9486 )

### DISCUSSION THREAD

**Discussion by qbox4u** [Bodmer/TFT_eSPI#2630] [Compatible displays and setup files]
Solution:
Add new generic setup file for Wemos D1 R32 in the folder User_Setups

Tested: OK

//***************************new setup file ******************************
//Setup for Wemos D1 R32  with 3.5inch TFT LCD shield
//
// https://nl.aliexpress.com/item/1005001928029759.html?
// This board is using the paralel interface !!!!
//
// The reason for not working is that the Wemos esp32 board connect CS,DC ands reset signals to 
// a2 a3 and a4 , but these pins are only for "INPUT!!!! use
// You need to modify the Wemos D1 R32 board as follow:
// Boot issue:
// Add 330 Ohm resistor  .           .==> https://github.com/espressif/arduino-esp32/issues/7562
// White screen issue:
// Connect gpio_15,33,32 to A1,2,3   .==>https://forum.arduino.cc/t/help-mcufriend-3-5-tft-lcd-480x320-esp32-solved/559854/5
//                                       https://github.com/Bodmer/TFT_eSPI/blob/master/docs/ESP32%20UNO%20board%20mod/ESP32%20UNO%20board%20mod.jpg

#define USER_SETUP_INFO "WemosD1R32"

////////////////////////////////////////////////////////////////////////////////////////////
// Interface
////////////////////////////////////////////////////////////////////////////////////////////
#define TFT_PARALLEL_8_BIT

////////////////////////////////////////////////////////////////////////////////////////////
// Display driver type
////////////////////////////////////////////////////////////////////////////////////////////
#define ILI9486_DRIVER

//#define TFT_INVERSION_OFF

#define TFT_CS    33  // Chip select control pin AFTER board modification
#define TFT_DC    15  // Data Command control pin AFTER board modification
#define TFT_RST   32  // Reset pin AFTER board modification
#define TFT_WR     4
#define TFT_RD     2

// PIO requires these to be sequentially increasing - do not change
#define TFT_D0   12
#define TFT_D1   13
#define TFT_D2   26
#define TFT_D3   25
#define TFT_D4   17
#define TFT_D5   16
#define TFT_D6   27
#define TFT_D7   14
//*/

#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT

// #define SPI_FREQUENCY  20000000
#define SPI_FREQUENCY  27000000
// #define SPI_FREQUENCY  40000000
// #define SPI_FREQUENCY  80000000

// Optional reduced SPI frequency for reading TFT
#define SPI_READ_FREQUENCY  16000000

#define SPI_TOUCH_FREQUENCY  2500000

**Comment by Bodmer** [comment]
There are lots of potential setups, so it is expected that an existing one is used as a template.

The addition of the 330 R resistor is an interesting idea, but will load the pin if used for other purposes. I have used a higher value resistor pulling down to 0V in the past, ISTR 10K was OK.
